backend/src/main/java/org/eclipse/tractusx/puris/backend/masterdata/logic/adapter/PartTypeInformationSammMapper.java

```diff
@@ -0,0 +1,119 @@
+        samm.setCatenaXId(material.getMaterialNumberCx());
+        samm.getPartTypeInformation().setManufacturerPartId(material.getOwnMaterialNumber());
+        samm.getPartTypeInformation().setNameAtManufacturer(material.getName());
+
+        return samm;
+    }
+
+    /**
+     * This method will accept a PartTypeInformation SAMM from a specific supplier partner,
+     * that is meant to update the information on a specific material.
+     *
+     * @param material          the material
+     * @param samm              the SAMM
+     * @param sendingPartner    the partner, you received the SAMM from
+     */
+    public void updateMaterialFromSamm(@NonNull Material material, @NonNull PartTypeInformationSAMM samm, @NonNull Partner sendingPartner) {
+        try {
+            var mpr = mprService.find(material, sendingPartner);
+            String partId = samm.getPartTypeInformation().getManufacturerPartId();
+            String nameAtManufacturer = samm.getPartTypeInformation().getNameAtManufacturer();
+            String cxId = samm.getCatenaXId();
+            // SAMM validity check
+            Objects.requireNonNull(mpr, "Missing Material Partner Relation");
+            Objects.requireNonNull(samm.getPartTypeInformation(), "Missing PartTypeInformation");
+            Objects.requireNonNull(partId, "Missing PartId");
+            Objects.requireNonNull(nameAtManufacturer, "Missing nameAtManufacturer");
+            Objects.requireNonNull(cxId, "Missing CatenaXId from partner");
+
+            // Receive update for Partner Material Number.
+            if (!partId.equals(mpr.getPartnerMaterialNumber())) {
+                // Notification if partner changed the Material Number for his product
+                if (mpr.getPartnerMaterialNumber() != null) {
+                    // This should never happen, because partners are not expected to change the materialNumber
+                    // of an existing material
+                    log.warn("Replacing previous Partner-MaterialNumber " + mpr.getPartnerMaterialNumber() + " with " +
+                        "new PartId from SAMM: " + partId + " by Partner " + sendingPartner.getBpnl());
+                }
+                mpr.setPartnerMaterialNumber(partId);
+            }
+
+            // Receive update for Partner's Name for Material
+            if (!nameAtManufacturer.equals(mpr.getNameAtManufacturer())) {
+                // Notification if partner changed the Name for his product
+                if (mpr.getNameAtManufacturer() != null) {
+                    log.warn("Replacing previous Name at Manufacturer " + mpr.getNameAtManufacturer() + " with " +
+                        "new Name from SAMM: " + nameAtManufacturer + " by Partner " + sendingPartner.getBpnl());
+                }
+                mpr.setNameAtManufacturer(nameAtManufacturer);
+            }
+
+            if (!cxId.equals(mpr.getPartnerCXNumber())) {
```

Comment:
The method, we are looking at here, is only meant to be used for the case that a supplier partner wants to create a PartType Samm for one of his customer partners. 
I was not aware of the fact that we also have to cover the situation for a case where a customer partner needs to create a PartType Samm for one of his suppliers. I thought that PartType Information was only passed on from a supplier towards a customer, not the other way around.

Reply:
You're right: A customer won't pass the samm to a supplier, but he will request it from the supplier and process it. Following the javadoc "This method will accept a PartTypeInformation SAMM from a specific supplier partner, that is meant to update the information on a specific material.", I assume we're in the same context (customer proccesses supplier's information).

With my bullets I only wanted to bring the sequence to the context (supplier already created a twin with partTypeInformation, customer now wants to update information -> especially the cx id)

Reply:
Ah, ok. I probably misunderstood that :)

Reply:
Ah wow, I also read the if statement wrong: it warns, if the cxNumber has already been set previously for a supplier. That's great! Thought it would also warn if it has not been set. So, I misread the code! :o